test_load_zip_with_local_header_extra_field failure on s390x

Greetings,

While integrating trx-python 0.4.0 in Debian sid, our continuous integration caught a regression on s390x (maybe also on other big endian platforms, but that's the only one tested):

https://ci.debian.net/packages/p/python-trx-python/testing/s390x/69895533/

The interesting part of the test log looks like:
```
 84s             trx = tmm.load_from_zip(trx_path)
 84s >           np.testing.assert_array_almost_equal(trx.streamlines._data, positions)
 84s E           AssertionError: 
 84s E           Arrays are not almost equal to 6 decimals
 84s E           
 84s E           Mismatched elements: 6 / 6 (100%)
 84s E           Max absolute difference among violations: 6.
 84s E           Max relative difference among violations: 1.
 84s E            ACTUAL: memmap([[4.6006e-41, 8.9683e-44, 2.3049e-41],
 84s E                   [4.6007e-41, 5.7487e-41, 6.8966e-41]], dtype='<f4')
 84s E            DESIRED: array([[1., 2., 3.],
 84s E                  [4., 5., 6.]], dtype=float32)
 84s 
 84s trx/test_memmap.py:609: AssertionError
```

I'm not sure exactly where the issue lies.  It could be a need to update the reference data for big endian platform, but the looks of the result feels more like a need to adjust the logic somewhere for big endian platforms appropriately.

I'm not expecting a lot of our users to run big endian platforms and intend to make this issue non-blocking for a migration to testing, so no pressure.  But if you feel this could be symptomatic of a more problematic issue, please don't hesitate to say so.

Have a nice day,  :)
Étienne.

--- a/trx/tests/test_memmap.py
+++ b/trx/tests/test_memmap.py
@@ -523,10 +523,12 @@ def test_load_zip_with_local_header_extra_field():
             local_info = []
             extra = b"\x00\x00\x04\x00TEST"  # 8-byte extra field
 
+            # TRX stores arrays little-endian on disk; serialize the fixture
+            # explicitly so it is valid on big-endian hosts too (issue #113).
             for name, data in [
                 ("header.json", json.dumps(header).encode()),
-                ("positions.3.float32", positions.tobytes()),
-                ("offsets.uint64", offsets.tobytes()),
+                ("positions.3.float32", positions.astype("<f4").tobytes()),
+                ("offsets.uint64", offsets.astype("<u8").tobytes()),
             ]:
                 offset = f.tell()
                 fname = name.encode()

--- a/trx/tests/test_memmap.py
+++ b/trx/tests/test_memmap.py
@@ -523,8 +523,6 @@ def test_load_zip_with_local_header_extra_field():
             local_info = []
             extra = b"\x00\x00\x04\x00TEST"  # 8-byte extra field
 
-            # TRX stores arrays little-endian on disk; serialize the fixture
-            # explicitly so it is valid on big-endian hosts too (issue #113).
             for name, data in [
                 ("header.json", json.dumps(header).encode()),
                 ("positions.3.float32", positions.astype("<f4").tobytes()),

